### PR TITLE
[CAKE-481] Enity related content and impact footer integration

### PR DESCRIPTION
**Pull request by ArturKlajnerok**
https://wikia-inc.atlassian.net/browse/CAKE-481

Adjusted the data produced by `CakeRelatedContentService/getContentRelatedTo` so we can use it as data source for impact footer.

//cc @Wikia/cake @pauloslund

### CODE REVIEW

File: extensions/wikia/Recirculation/services/CakeRelatedContentService.php [hunk 1]
```diff
@@ -141,13 +146,36 @@ private function formatTitle(Content $content) {
...
+	private function getAuthor($metadata) {
+		if (array_key_exists("authorName", $metadata)) {
+			return $metadata["authorName"];
+		} else {
+			return "";
+		}
+
+	}
+
+	private function getRecirculationContentType($contentType) {
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/services/CakeRelatedContentService.php:165]: MAJOR Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
- MAJOR 2 major
- MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [CakeRelatedContentService.php#L24](https://github.com/Wikia/app/blob/85f894ab28efa1bf6a3b1bd30b00f0c25af44827/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L24): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
2. MINOR [CakeRelatedContentService.php#L37](https://github.com/Wikia/app/blob/85f894ab28efa1bf6a3b1bd30b00f0c25af44827/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L37): Define a constant instead of duplicating this literal "title" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
3. MINOR [CakeRelatedContentService.php#L38](https://github.com/Wikia/app/blob/85f894ab28efa1bf6a3b1bd30b00f0c25af44827/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L38): Define a constant instead of duplicating this literal "limit" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by nmonterroso** [comment]
LGTM

**Comment by pauloslund** [comment]
LGTM

File: extensions/wikia/Recirculation/services/CakeRelatedContentService.php [hunk 2]
```diff
@@ -141,13 +146,36 @@ private function formatTitle(Content $content) {
...
+		switch ($contentType) {
+			case "Discussion Thread":
+				return "discussions";
+			case "Fandom Article":
+				return "fandom";
+			case "Wiki Article":
+				return "wiki";
+			default:
+				return "undefined";
+		}
```
Comment (drsnyder) [inline extensions/wikia/Recirculation/services/CakeRelatedContentService.php:175]: It seems like this should be a method on the `$content` objects. Something like `$content->getSource()` might work.

**Pull request merged**